doc/holochain_101/src/zome/zome_functions.md

```diff
@@ -120,11 +120,146 @@ Imagine that there are many DNA instances running within one Container, and each
 - which function?
 - what arguments?
 
-Containers can implement whatever interfaces to perform these function calls they wish to, opening a wealth of opportunity. Holochain provides two reference Containers, one for [Nodejs](https://www.npmjs.com/package/@holochain/holochain-nodejs), and the other a [Rust built binary executable](https://github.com/holochain/holochain-rust/tree/develop/container). With the Rust built binary Container, interfaces for making function calls already includes HTTP and WebSockets. More details about Containers can be found in [another chapter](../container.md), it is simply important context for proceeding.
+Containers can implement whatever interfaces to perform these function calls they wish to, opening a wealth of opportunity. Holochain provides two reference Containers, one for [Nodejs](https://www.npmjs.com/package/@holochain/holochain-nodejs), and the other a [Rust built binary executable](https://github.com/holochain/holochain-rust/tree/develop/container). With the Rust built binary Container, interfaces for making function calls already includes HTTP and WebSockets. More details about Containers can be found in [another chapter](../containers.md), it is simply important context for proceeding.
 
 When a call to a Zome function is being made from the Container, it first passes the arguments to Holochain. Before making the function call, Holochain will check the validity of the request, and fail if necessary. If the request is deemed valid, Holochain will mount the WASM code for a Zome using its' WASM interpreter, and then make a function call into it, giving it the arguments given to it in the request. When it receives the response from the WASM, it will then pass that return value as the response to the request. This may sound complex, but that's just what's going on internally, actually using it with an HDK and a Container is easy.
 
 
 ## Building in Rust: Zome Functions
 
-Coming soon...
+So far, in [entry type definitions](./entry_type_definitions.md) and [genesis](./genesis.md), the most complex example of `define_zome!` was still very simple, and didn't include any functions:
+
+```rust
+...
+
+#[derive(Serialize, Deserialize, Debug, DefaultJson)]
+struct Post {
+    content: String,
+    date_created: String,
+}
+
+define_zome! {
+    entries: [
+        entry!(
+            name: "post",
+            description: "A blog post entry which has an author",
+            sharing: Sharing::Public,
+            native_type: Post,
+            validation_package: || {
+                ValidationPackageDefinition::Entry
+            },
+            validation: |_post: Post, _validation_data: ValidationData| {
+                Ok(())
+            }
+        )
+    ]
+
+    genesis: || {
+        Ok(())
+    }
+
+    functions: {}
+}
+```
+
+`functions` is where the Capabilities, and function declarations will be made.
+
+### Adding a Capability
```

Comment:
This will no longer be correct after #791 gets merged.  As a matter of process I'm happy to do the update to this text as part of that PR if this gets merged first.

Reply:
We can work together to make the relevant updates @zippy sounds good